File: docs/current/guides/snippets/use-services/use-hostnames-2/main.go
package main

import (
	"context"
	"fmt"
	"os"

	"dagger.io/dagger"
)

func main() {
	ctx := context.Background()

	// create Dagger client
	client, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stderr))

	if err != nil {
		panic(err)
	}
	defer client.Close()

	// get hostname of service container
	val, err := client.Container().
		From("alpine").
		WithExec([]string{"hostname"}).
		Stdout(ctx)

	if err != nil {
		panic(err)
	}

	fmt.Println(val)
}

File: docs/current/guides/snippets/use-services/use-hostnames-2/main.py
import sys

import anyio
import dagger


async def main():
    # create Dagger client
    async with dagger.Connection(dagger.Config(log_output=sys.stderr)) as client:
        # get hostname of service container
        val = await client.container().from_("alpine").with_exec(["hostname"]).stdout()

    print(val)


anyio.run(main)
